src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -59,7 +59,8 @@ public static SimilarityFunction fromString(String similarityFunction) {
 
   public static CollectionSettings getCollectionSettings(
       TableMetadata table, ObjectMapper objectMapper) {
-    String collectionName = table.getName().asCql(true);
+    // [jsonapi#639]: get internal name to avoid quoting of case-sensitive names
+    String collectionName = table.getName().asInternal();
```

Comment:
This is the actual fix: prevents name from being surrounded by double-quotes (needed for CQL usage, but not in JSON)